src/metric-engine/src/engine.rs

```diff
@@ -196,12 +205,13 @@ struct MetricEngineInner {
     mito: MitoEngine,
     metadata_region: MetadataRegion,
     data_region: DataRegion,
-    // TODO(ruihang): handle different catalog/schema
-    /// Map from physical table name to table id.
-    physical_tables: RwLock<HashMap<String, TableId>>,
+    /// Mapping from physical table id to its logical table ids
+    physical_tables: RwLock<HashMap<RegionId, HashSet<RegionId>>>,
```

Comment:
Maybe `physical_tables` and `physical_columns` should use the same lock to avoid deadlock.

Reply:
changed in https://github.com/GreptimeTeam/greptimedb/pull/2726/commits/e6fb4cdc5214e95fdd112283943271e901911361